modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/ClusterMetricsImpl.java

```diff
@@ -270,27 +409,27 @@ public ClusterMetricsImpl(GridKernalContext ctx, GridLocalMetrics vmMetrics, lon
 
     /** {@inheritDoc} */
     @Override public long getNonHeapMemoryCommitted() {
-        return vmMetrics.getNonHeapMemoryCommitted();
+        return nonHeapCommitted.value();
     }
 
     /** {@inheritDoc} */
     @Override public long getNonHeapMemoryMaximum() {
-        return vmMetrics.getNonHeapMemoryMaximum();
+        return nonHeapMax.value();
     }
 
     /** {@inheritDoc} */
     @Override public long getNonHeapMemoryTotal() {
-        return vmMetrics.getNonHeapMemoryMaximum();
+        return nonHeapMax.value();
     }
 
     /** {@inheritDoc} */
     @Override public long getUpTime() {
-        return vmMetrics.getUptime();
+        return upTime.value();
     }
 
     /** {@inheritDoc} */
     @Override public long getStartTime() {
-        return vmMetrics.getStartTime();
+        return rt.getStartTime();
```

Comment:
Why different approaches are used for `getUpTime()` and `getStartTime()`? Both from the same MXBean, both used only once in `ClusterMetricsImpl`. Wouldn't it be better to get start time from the metrics registry too?

Reply:
`startTime` is a constant that can be derived from `upTime` so we don't need it in `MetricRegistry`.